lib/src/source_visitor.dart

```diff
@@ -716,37 +716,50 @@ class SourceVisitor extends ThrowingAstVisitor {
       //       parameter,
       //     ) : super();
       space();
+
+      var isOptional =
+          node.parameters.parameters.last.kind != ParameterKind.REQUIRED;
+      if (node.initializers.length > 1) {
+        _writeText(isOptional ? " " : "  ", node.separator.offset);
+      }
+
+      // ":".
+      token(node.separator);
+
+      space();
+
+      builder.indent(6);
     } else {
       // Shift the itself ":" forward.
       builder.indent(Indent.constructorInitializer);
 
       // If the parameters or initializers split, put the ":" on its own line.
```

Comment:
this comment feels strange to me.

Every `hasTrailingComma` would have split parameters, but not all split parameters or initializers are caused by `hasTrailingComma`.

Maybe "// Every parameter list with a trailing comma will have split initializers which puts the ":" on it's own line"

Reply:
This section of code is for when the parameter list does *not* have a trailing comma, so I'm not sure we need to mention that here.

Reply:
ugh yeah, I'm even more confused....

How do we know the parameters or intializers are split when we get here?

Reply:
Sorry, this makes more sense if you're more familiar with the codebase. The `split()` function means "*allow* it to split here" but doesn't require it. It's inserting a potential split, not a mandatory one. That's why the preceding comment says "if". The `split()` call works for both cases.